### PR TITLE
[DPE-1761] Juju 3

### PR DESCRIPTION
**Pull request by dragomirp**
## Issue
We should run integration tests against both juju 3 and juju 2

## Solution
* Add support for running integration test on both juju versions

### CODE REVIEW

File: .github/workflows/ci.yaml
Comment (dragomirp) [inline .github/workflows/ci.yaml:1]: Copy paste from the other charms

File: src/relations/db.py [hunk 1]
```diff
@@ -258,7 +258,6 @@ def _on_relation_joined(self, join_event: RelationJoinedEvent):
         except (PostgreSQLCreateDatabaseError, PostgreSQLCreateUserError):
             err_msg = f"failed to create database or user for {self.relation_name}"
             logger.error(err_msg)
-            join_event.fail()
```
Comment (dragomirp) [inline src/relations/db.py:261]: There's no fail method in [EventBase](https://github.com/canonical/operator/blob/2.5.0/ops/framework.py#L176). Most probably a copy paste error.

File: tests/integration/helpers/postgresql_helpers.py [hunk 2]
```diff
@@ -192,23 +192,3 @@ def get_unit_address(ops_test: OpsTest, unit_name: str) -> str:
...
-    Returns:
-        the command output if it succeeds, otherwise raises an exception.
-    """
-    complete_command = f"run --unit {unit_name} -- {command}"
-    return_code, stdout, _ = await ops_test.juju(*complete_command.split())
-    if return_code != 0:
-        raise Exception(
-            "Expected command %s to succeed instead it failed: %s", command, return_code
-        )
-    return stdout
```
Comment (dragomirp) [inline tests/integration/helpers/postgresql_helpers.py:214]: Moved to avoid circular imports in the helper modules.

**Review by marceloneppel — APPROVED** [review]
LGTM!

**Review by taurus-forever — APPROVED** [review]

**Pull request merged**